### ISSUE TITLE
IDE0055 for indented comment

### ISSUE THREAD

**Issue by brnls** [dotnet/roslyn#66478]
**Version Used**: 

7.0.102
Windows

**Steps to Reproduce**:

Using source code here: https://github.com/brnls/IDE0055-Bug

1. Convert line endings in `Program.cs` to `LF`
2. Run `dotnet build`

**Diagnostic Id**:
IDE0055 

**Expected Behavior**:
No formatting errors when running `dotnet build`

**Actual Behavior**:
`dotnet build` produces formatting errors. 

The presence of the indented comment seems to trigger this. If I remove it or unindent it, IDE0055 doesn't trigger.
Also I tried it on Ubuntu with the same dotnet sdk version. It has the reverse behavior where if I convert line endings to CRLF it will fail to build, but if line endings are LF it is fine.

It seems like for any line with a comment that is indented, it triggers if the line doesn't end in `Environment.NewLine`.

**Comment by randyshoopman** [comment]
I'm seeing this same issue when running a build in Github Actions (Ubuntu, 22.04.2).   Our code is stored in github with LFs and is translated to CRLF when cloned onto our developer computers running Windows.  

.gitattributes:
```
# Auto detect text files and perform LF normalization
* text=auto
```

So when running locally in Windows we see no formatting errors (IDE0055) but then those errors appear when we build with Ubuntu in Github Actions, presumably because the LF aren't converted to CRLF in Linux.

**Comment by sharwell** [comment]
@randyshoopman Can you verify that one of the following is applied?

1. _Neither_ **.gitattributes** or **.editorconfig** specifies the end of line sequence
2. _Both_ **.gitattributes** and **.editorconfig specifies the end of line sequence

My recommendation is to use the first option.
